cosmic/data_source_cosmic_common_schema.go

```diff
@@ -0,0 +1,55 @@
+package cosmic
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceFiltersSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Required: true,
+		ForceNew: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"name": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"value": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+			},
+		},
+	}
+}
+
+// matchFilters returns true or false depending if the passed filters are matched.
+func matchFilters(v interface{}, filters *schema.Set) (bool, error) {
+	var j map[string]interface{}
+	t, _ := json.Marshal(v)
+	json.Unmarshal(t, &j)
+
+	for _, f := range filters.List() {
+		m := f.(map[string]interface{})
+
+		r, err := regexp.Compile(m["value"].(string))
+		if err != nil {
+			return false, fmt.Errorf("Invalid regex: %s", err)
+		}
+
+		field, ok := j[m["name"].(string)].(string)
+		if !ok {
+			return false, fmt.Errorf("Invalid field name: %s", m["name"])
+		}
```

Comment:
I would rename `field` to `value` and first do this check, then create the `regexp` (which is usually stored in a var called `re`) and the try to match it.

Reply:
Fixed